### PR TITLE
sdl2_mixer: make libmodplug required instead of optional

### PR DESCRIPTION
**Pull request by frranck**
ref https://github.com/Homebrew/homebrew-core/pull/20746

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by frranck** [comment]
ref https://github.com/Homebrew/homebrew-core/pull/20746

File: Formula/sdl2_mixer.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ class Sdl2Mixer < Formula
   depends_on "flac" => :optional
   depends_on "fluid-synth" => :optional
   depends_on "libmikmod" => :optional
-  depends_on "libmodplug" => :optional
+  depends_on "libmodplug"
```
Comment (ilovezfs) [inline Formula/sdl2_mixer.rb]: needs to go above libvorbis

**Comment by ilovezfs** [comment]
also please bump the revision

**Comment by ilovezfs** [comment]
rebuild is not the same as revision. you shouldn't modify the bottle block manually. revision goes at formula global scope.

**Comment by frranck** [comment]
@ilovezfs you mean :revision ?

**Comment by ilovezfs** [comment]
nope that's a component of urls

`revision 1`

File: Formula/sdl2_mixer.rb [hunk 2]
```diff
@@ -2,12 +2,12 @@ class Sdl2Mixer < Formula
   desc "Sample multi-channel audio mixer library"
   homepage "https://www.libsdl.org/projects/SDL_mixer/"
   url "https://www.libsdl.org/projects/SDL_mixer/release/SDL2_mixer-2.0.2.tar.gz"
+  revision 1
```
Comment (ilovezfs) [inline Formula/sdl2_mixer.rb]: goes under sha256

**Comment by ilovezfs** [comment]
Thanks. Please squash this down to one commit with the commit subject.
```
sdl2_mixer: depend on libmodplug
```

**Comment by ilovezfs** [comment]
actually make it
```
sdl2_mixer: make libmodplug required instead of optional
```

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Review by ilovezfs — APPROVED** [review]

File: Formula/sdl2_mixer.rb [hunk 3]
```diff
@@ -30,14 +31,13 @@ def install
       --disable-music-flac-shared
       --disable-music-midi-fluidsynth-shared
       --disable-music-mod-mikmod-shared
-      --disable-music-mod-modplug-shared
```
Comment (ilovezfs) [inline Formula/sdl2_mixer.rb]: actually we want to keep this disabled so that the linkage is explicit

Comment (frranck) [reply Formula/sdl2_mixer.rb]: I'm not sure I understand, this defines the -DMODPLUG_DYNAMIC, without this we need to compile libmodplug statically with mrboom, but we don't have a libmodplug.a to do that.

Comment (ilovezfs) [reply Formula/sdl2_mixer.rb]: The difference is not whether it's static or dynamic but whether the path to the dylib is explicit or not.

Comment (ilovezfs) [reply Formula/sdl2_mixer.rb]: With `--disable-music-mod-modplug-shared`, here's what you get:
```
iMac-TMP:~ joe$ brew linkage sdl2_mixer
System libraries:
  /System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox
  /System/Library/Frameworks/AudioUnit.framework/Versions/A/AudioUnit
  /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
  /System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices
  /usr/lib/libSystem.B.dylib
Homebrew libraries:
  /usr/local/opt/libmodplug/lib/libmodplug.1.dylib (libmodplug)
  /usr/local/opt/libvorbis/lib/libvorbis.0.dylib (libvorbis)
  /usr/local/opt/libvorbis/lib/libvorbisfile.3.dylib (libvorbis)
  /usr/local/opt/sdl2/lib/libSDL2-2.0.0.dylib (sdl2)
iMac-TMP:~ joe$ 
```

Comment (frranck) [reply Formula/sdl2_mixer.rb]: But it does compile sdl2_mixer differently if you look at music_modplug.c, is this OK?

Comment (ilovezfs) [reply Formula/sdl2_mixer.rb]: Can you test whether `mrboom` works as expected if you compile with `--disable-music-mod-modplug-shared`?

Comment (ilovezfs) [reply Formula/sdl2_mixer.rb]: :+1: :-1: ?

Comment (frranck) [reply Formula/sdl2_mixer.rb]: It works fine with SDL2_mixer --with-libmodplug

Comment (ilovezfs) [reply Formula/sdl2_mixer.rb]: Cool.

Comment (frranck) [reply Formula/sdl2_mixer.rb]: So since --disable-music-mod-modplug-shared was a default it shouldn't change right ?

Comment (ilovezfs) [reply Formula/sdl2_mixer.rb]: correct

Comment (ilovezfs) [reply Formula/sdl2_mixer.rb]: And I assume you tested it without the `--with-libmodplug`?

Comment (ilovezfs) [reply Formula/sdl2_mixer.rb]: By which I mean, are you sure the new dependency is actually required?

Comment (ilovezfs) [reply Formula/sdl2_mixer.rb]: @frranck ?

Comment (frranck) [reply Formula/sdl2_mixer.rb]: yeah sure,
https://github.com/Javanaise/mrboom-libretro/blob/master/common.cpp#L348

Comment (ilovezfs) [reply Formula/sdl2_mixer.rb]: ok

**Comment by ilovezfs** [comment]
@frranck Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**